README.rst

```diff
@@ -1001,6 +1092,65 @@ For example, sending recovery notifications.
 This signal is emitted when the threshold value of a ``Metric`` defined in
 alert settings is crossed.
 
+``pre_metric_write``
+~~~~~~~~~~~~~~~~~~~~
+
+**Path**: ``openwisp_monitoring.monitoring.signals.pre_metric_write``
+
+**Arguments**:
+
+- ``metric``: ``Metric`` object whose data shall be stored in timeseries database
+- ``values``: metric data that shall be stored in the timeseries database
+
+This signal is emitted for every metric **just before** a request is made to
+timeseries database to store that metric's latest data.
+
+``post_metric_write``
+~~~~~~~~~~~~~~~~~~~~~
+
+**Path**: ``openwisp_monitoring.monitoring.signals.post_metric_write``
+
+**Arguments**:
+
+- ``metric``: ``Metric`` object whose data is being stored in timeseries database
+- ``values``: metric data that is being stored in the timeseries database
+
+This signal is emitted for every metric **just after** a task is started to store data in the
+timeseries database (run in the background).
```

Comment:
after the write operation is sent to the timeseries database. Since the write operation is run in the background and may be retried on failures, using the ``post_metric_write`` signal does not guarantee the receiver function will be called after the data has been written, only after the request to write it has been sent.

Reply:
The second line seems a bit confusing to me (the part starting from *only).

Reply:
feel free to improve it if you find it confusing, you can also seek feedback from other people in the community, my english is not so great after all so it's very good to seek counsel from others

Reply:
How about below one, we simply break the sentence
Since the write operation is run in the background and may be retried on failures, using the post_metric_write signal does not guarantee the receiver function will be called after the data has been written. It only indicates that the request to write data has been sent.